diff --git a/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h b/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h
index f40968746..ed81689f9 100644
--- a/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h
+++ b/modules/juce_audio_plugin_client/utility/juce_CheckSettingMacros.h
@@ -74,7 +74,7 @@
  #define JucePlugin_Build_RTAS 0
 #endif
 
-#if ! (defined (_MSC_VER) || defined (__APPLE_CPP__) || defined (__APPLE_CC__))
+#if ! (defined (_MSC_VER) || defined(__MINGW32__) || defined (__APPLE_CPP__) || defined (__APPLE_CC__))
  #undef JucePlugin_Build_VST3
  #define JucePlugin_Build_VST3 0
 #endif
